src/libsync/configfile.cpp

```diff
@@ -647,7 +648,23 @@ void ConfigFile::setProxyType(int proxyType,
         settings.setValue(QLatin1String(proxyPortC), port);
         settings.setValue(QLatin1String(proxyNeedsAuthC), needsAuth);
         settings.setValue(QLatin1String(proxyUserC), user);
-        settings.setValue(QLatin1String(proxyPassC), pass.toUtf8().toBase64());
+
+        if (pass.isEmpty()) {
+            settings.remove(QLatin1String(proxyPassC));
+
+            auto *job = new QKeychain::DeletePasswordJob(Theme::instance()->appName());
+            job->setInsecureFallback(false);
+            job->setKey(keychainProxyPasswordKey());
+            job->setAutoDelete(true);
+            job->start();
+        } else {
+            // Write password to keychain
+            auto *job = new KeychainChunk::WriteJob(keychainProxyPasswordKey(), pass.toUtf8());
+            if (job->startAwait()) {
```

Comment:
Hm, so we potentially block at startup... couldn't we do that async? "No" is of course a proper answer if motivated, it's just that our startup time isn't great already so if there's a better option I'm all for it.

Reply:
Checked it: The migration occurred twice in the log, so I finally decided for "no" even thought I'd have liked "yes" ^^

Reply:
OK